rules/utils/boolean.js

```diff
@@ -71,6 +73,26 @@ function isBooleanNode(node) {
 		return isBooleanNode(parent);
 	}
 
+	if (scope && parent.type === 'VariableDeclarator') {
+		const reference = findReference(scope, parent.id);
```

Comment:
The `parent.id` can be other things, not only `Identifier`, we need tests.

Reply:
Could you suggest me some tests that I need to add?

Reply:
```js
const {} = foo;
const [] = foo;
```